backend/src/main/java/com/bakdata/conquery/models/externalservice/ResultType.java

```diff
@@ -24,27 +26,40 @@
 import org.apache.arrow.vector.types.pojo.ArrowType;
 import org.apache.arrow.vector.types.pojo.Field;
 import org.apache.arrow.vector.types.pojo.FieldType;
+import org.apache.poi.ss.usermodel.Cell;
+import org.apache.poi.ss.usermodel.CellStyle;
 
 @JsonTypeInfo(use = JsonTypeInfo.Id.CUSTOM, property = "type")
 @CPSBase
-public interface ResultType {
+public abstract class ResultType {
 
-    default String printNullable(PrintSettings cfg, Object f) {
+    public String printNullable(PrintSettings cfg, Object f) {
         if (f == null) {
             return "";
         }
         return print(cfg, f);
     }
 
-    default String print(PrintSettings cfg, @NonNull Object f) {
+    protected String print(PrintSettings cfg, @NonNull Object f) {
         return f.toString();
     }
 
-    Field getArrowFieldType(ResultInfo info, PrintSettings settings);
+    public abstract Field getArrowFieldType(ResultInfo info, PrintSettings settings);
 
-    String typeInfo();
+    public void writeExcelCell(ResultInfo info, PrintSettings settings, Cell cell, Object value, Map<String, CellStyle> styles){

```

Comment:
Die Klasse mischt jetzt Arrow und Excel und Csv, macht das wirklich Sinn? Könntest du die methoden nicht verscheiebn wo sie besser passen?

Reply:
Ich könnte die Methoden rausnehmen und dann in einem großen if-else-if verbundlen

Reply:
Ich schau mir das nachher auch mal an was da genau passiert, vlt kann man das an entsprechende stellen ziehen

Reply:
Oder einer Map

Reply:
Habs in einer Map getan